src/Http/Http.Extensions/test/RequestDelegateGenerator/Baselines/MapAction_JsonBodyOrService_HandlesBothJsonAndService.generated.txt

```diff
@@ -8,51 +8,15 @@
 //------------------------------------------------------------------------------
 #nullable enable
 
-namespace Microsoft.AspNetCore.Builder
+namespace System.Runtime.CompilerServices
 {
     %GENERATEDCODEATTRIBUTE%
-    internal sealed class SourceKey
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = true)]
+    file sealed class InterceptsLocationAttribute : Attribute
```

Comment:
What happens if another source generator adds this same `file sealed class InterceptsLocationAttribute` to the project? For example - if I enable both RDG and the ConfigurationBinder source generator.

Reply:
Since the type is file scoped, it'll be fine to have multiple definitions in each generator as each will produce a unique type definition (e.g. `System.Runtime.CompilerServices.<GeneratedFileName_g>FBAC_InterceptsLocationAttribute`).

Things will get interesting if generators aren't cautious about making them file private (as opposed to say internal) but we should fine here.

cc: @RikkiGibson At minimum, we probably want to document the requirement for `InterceptsLocationAttribute` definitions to be file private. Not sure if we can add a compiler check/analyzer as well.